### PR TITLE
Fix a few more SSR problems

### PR DESCRIPTION
**Pull request by xvaara**
Tested SSR render with bootstrap-vue-3. Found these.

### CODE REVIEW

**Comment by GeoSot** [comment]
Thank you @xvaara  :heart:

File: js/src/dom/event-handler.js [hunk 1]
```diff
@@ -150,7 +150,7 @@ function normalizeParams(originalTypeEvent, handler, delegationFn) {
 }
 
 function addHandler(element, originalTypeEvent, handler, delegationFn, oneOff) {
-  if (typeof originalTypeEvent !== 'string' || !element) {
+  if (typeof originalTypeEvent !== 'string' || !element || !element.addEventListener) {
```
Comment (GeoSot) [inline js/src/dom/event-handler.js:153]: @xvaara  can you define the reason of this change?

File: js/src/util/index.js [hunk 2]
```diff
@@ -321,7 +321,7 @@ const getWindow = () => {
  * @return {document|{}} The proper element
  */
 const getDocument = () => {
-  return typeof document !== 'undefined' ? document : {}
+  return typeof document !== 'undefined' ? document : { documentElement: {} }
```
Comment (GeoSot) [inline js/src/util/index.js:324]: @xvaara  can you define the reason of this change, too?

**Comment by GeoSot** [comment]
Closing it as not answered. The focusTrap changes transferred to #34989

**Comment by xvaara** [comment]
I did answer... ?

**Pull request closed without merging**